File: gamslib/spatequ/spatequ-scalar.py
#  MCP written by GAMS Convert at 12/13/18 11:30:36
#  
#  Equation counts
#      Total        E        G        L        N        X        C        B
#         30       18       12        0        0        0        0        0
#  
#  Variable counts
#                   x        b        i      s1s      s2s       sc       si
#      Total     cont   binary  integer     sos1     sos2    scont     sint
#         30       30        0        0        0        0        0        0
#  FX      0        0        0        0        0        0        0        0
#  
#  Nonzero counts
#      Total    const       NL      DLL
#         84       84        0        0


from pyomo.environ import *
from pyomo.mpec import Complementarity

model = m = ConcreteModel()


m.x1 = Var(within=Reals,bounds=(None,None),initialize=101)
m.x2 = Var(within=Reals,bounds=(None,None),initialize=195.272727272727)
m.x3 = Var(within=Reals,bounds=(None,None),initialize=64)
m.x4 = Var(within=Reals,bounds=(None,None),initialize=35.9090909090909)
m.x5 = Var(within=Reals,bounds=(None,None),initialize=90)
m.x6 = Var(within=Reals,bounds=(None,None),initialize=154.272727272727)
m.x7 = Var(within=Reals,bounds=(None,None),initialize=65.5)
m.x8 = Var(within=Reals,bounds=(None,None),initialize=120.363636363636)
m.x9 = Var(within=Reals,bounds=(None,None),initialize=134.5)
m.x10 = Var(within=Reals,bounds=(None,None),initialize=160.227272727273)
m.x11 = Var(within=Reals,bounds=(None,None),initialize=55)
m.x12 = Var(within=Reals,bounds=(None,None),initialize=104.863636363636)
m.x13 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x14 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x15 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x16 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x17 = Var(within=Reals,bounds=(0,None),initialize=35.5)
m.x18 = Var(within=Reals,bounds=(0,None),initialize=74.9090909090909)
m.x19 = Var(within=Reals,bounds=(0,None),initialize=35.0000000000001)
m.x20 = Var(within=Reals,bounds=(0,None),initialize=49.409090909091)
m.x21 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x22 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x23 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x24 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x25 = Var(within=Reals,bounds=(0,None),initialize=11)
m.x26 = Var(within=Reals,bounds=(0,None),initialize=11.6363636363636)
m.x27 = Var(within=Reals,bounds=(0,None),initialize=9)
m.x28 = Var(within=Reals,bounds=(0,None),initialize=8.63636363636364)
m.x29 = Var(within=Reals,bounds=(0,None),initialize=10)
m.x30 = Var(within=Reals,bounds=(0,None),initialize=10.6363636363636)

m.c1 = Complementarity(expr=( - m.x1 - 9*m.x25 == -200)

m.c2 = Complementarity(expr=( - m.x2 - 9*m.x26 == -300)

m.c3 = Complementarity(expr=( - m.x3 - 4*m.x27 == -100)

m.c4 = Complementarity(expr=( - m.x4 - 19*m.x28 == -200)

m.c5 = Complementarity(expr=( - m.x5 - 7*m.x29 == -160)

m.c6 = Complementarity(expr=( - m.x6 - 9*m.x30 == -250)

m.c7 = Complementarity(expr=( - m.x7 + 10.5*m.x25 == 50)

m.c8 = Complementarity(expr=( - m.x8 + 15.5*m.x26 == 60)

m.c9 = Complementarity(expr=( - m.x9 + 20.5*m.x27 == 50)

m.c10 = Complementarity(expr=( - m.x10 + 25.5*m.x28 == 60)

m.c11 = Complementarity(expr=( - m.x11 + 10.5*m.x29 == 50)

m.c12 = Complementarity(expr=( - m.x12 + 15.5*m.x30 == 60)

m.c13 = Complementarity(expr=( - m.x1 + m.x7 - m.x13 - m.x15 + m.x17 + m.x21 >= 0, m.x25 >= 0))

m.c14 = Complementarity(expr=( - m.x2 + m.x8 - m.x14 - m.x16 + m.x18 + m.x22 >= 0, m.x26 >= 0))

m.c15 = Complementarity(expr=( - m.x3 + m.x9 + m.x13 - m.x17 - m.x19 + m.x23 >= 0, m.x27 >= 0))

m.c16 = Complementarity(expr=( - m.x4 + m.x10 + m.x14 - m.x18 - m.x20 + m.x24 >= 0, m.x28 >= 0))

m.c17 = Complementarity(expr=( - m.x5 + m.x11 + m.x15 + m.x19 - m.x21 - m.x23 >= 0, m.x29 >= 0))

m.c18 = Complementarity(expr=( - m.x6 + m.x12 + m.x16 + m.x20 - m.x22 - m.x24 >= 0, m.x30 >= 0))

m.c19 = Complementarity(expr=(   m.x25 - m.x27 >= -2, m.x13 >= 0))

m.c20 = Complementarity(expr=(   m.x26 - m.x28 >= -3, m.x14 >= 0))

m.c21 = Complementarity(expr=(   m.x25 - m.x29 >= -2, m.x15 >= 0))

m.c22 = Complementarity(expr=(   m.x26 - m.x30 >= -3, m.x16 >= 0))

m.c23 = Complementarity(expr=( - m.x25 + m.x27 >= -2, m.x17 >= 0))

m.c24 = Complementarity(expr=( - m.x26 + m.x28 >= -3, m.x18 >= 0))

m.c25 = Complementarity(expr=(   m.x27 - m.x29 >= -1, m.x19 >= 0))

m.c26 = Complementarity(expr=(   m.x28 - m.x30 >= -2, m.x20 >= 0))

m.c27 = Complementarity(expr=( - m.x25 + m.x29 >= -2, m.x21 >= 0))

m.c28 = Complementarity(expr=( - m.x26 + m.x30 >= -3, m.x22 >= 0))

m.c29 = Complementarity(expr=( - m.x27 + m.x29 >= -1, m.x23 >= 0))

m.c30 = Complementarity(expr=( - m.x28 + m.x30 >= -2, m.x24 >= 0))

File: gamslib/spatequ/spatequ-scalar.jl
#  MCP written by GAMS Convert at 10/11/20 13:33:30
#  
#  Equation counts
#      Total        E        G        L        N        X        C        B
#         30       18       12        0        0        0        0        0
#  
#  Variable counts
#                   x        b        i      s1s      s2s       sc       si
#      Total     cont   binary  integer     sos1     sos2    scont     sint
#         30       30        0        0        0        0        0        0
#  FX      0        0        0        0        0        0        0        0
#  
#  Nonzero counts
#      Total    const       NL      DLL
#         84       84        0        0
#  Cannot handle MCP


using JuMP
using MathOptInterface
const MOI = MathOptInterface

model = m = Model()

@variable(m, x1, start=101.000000000004)
@variable(m, x2, start=195.272727272719)
@variable(m, x3, start=63.9999999999913)
@variable(m, x4, start=35.9090909091103)
@variable(m, x5, start=89.9999999999702)
@variable(m, x6, start=154.272727272719)
@variable(m, x7, start=65.4999999999948)
@variable(m, x8, start=120.363636363593)
@variable(m, x9, start=134.49999999997)
@variable(m, x10, start=160.227272727363)
@variable(m, x11, start=55.0000000000011)
@variable(m, x12, start=104.863636363592)
@variable(m, 0 <= x13, start=0)
@variable(m, 0 <= x14, start=0)
@variable(m, 0 <= x15, start=0)
@variable(m, 0 <= x16, start=0)
@variable(m, 0 <= x17, start=35.5000000000096)
@variable(m, 0 <= x18, start=74.9090909091256)
@variable(m, 0 <= x19, start=34.999999999969)
@variable(m, 0 <= x20, start=49.4090909091271)
@variable(m, 0 <= x21, start=0)
@variable(m, 0 <= x22, start=0)
@variable(m, 0 <= x23, start=0)
@variable(m, 0 <= x24, start=0)
@variable(m, 0 <= x25, start=11)
@variable(m, 0 <= x26, start=11.6363636363603)
@variable(m, 0 <= x27, start=9)
@variable(m, 0 <= x28, start=8.63636363636033)
@variable(m, 0 <= x29, start=10.0000000000036)
@variable(m, 0 <= x30, start=10.636363636364)

@constraint(m,  - x1 - 9*x25 == -200)

@constraint(m,  - x2 - 9*x26 == -300)

@constraint(m,  - x3 - 4*x27 == -100)

@constraint(m,  - x4 - 19*x28 == -200)

@constraint(m,  - x5 - 7*x29 == -160)

@constraint(m,  - x6 - 9*x30 == -250)

@constraint(m,  - x7 + 10.5*x25 == 50)

@constraint(m,  - x8 + 15.5*x26 == 60)

@constraint(m,  - x9 + 20.5*x27 == 50)

@constraint(m,  - x10 + 25.5*x28 == 60)

@constraint(m,  - x11 + 10.5*x29 == 50)

@constraint(m,  - x12 + 15.5*x30 == 60)

@constraint(m,  - x1 + x7 - x13 - x15 + x17 + x21 == 0)

@constraint(m,  - x2 + x8 - x14 - x16 + x18 + x22 == 0)

@constraint(m,  - x3 + x9 + x13 - x17 - x19 + x23 == 0)

@constraint(m,  - x4 + x10 + x14 - x18 - x20 + x24 == 0)

@constraint(m,  - x5 + x11 + x15 + x19 - x21 - x23 == 0)

@constraint(m,  - x6 + x12 + x16 + x20 - x22 - x24 == 0)

@constraint(m, x25 - x27 >= -2)

@constraint(m, x26 - x28 >= -3)

@constraint(m, x25 - x29 >= -2)

@constraint(m, x26 - x30 >= -3)

@constraint(m,  - x25 + x27 >= -2)

@constraint(m,  - x26 + x28 >= -3)

@constraint(m, x27 - x29 >= -1)

@constraint(m, x28 - x30 >= -2)

@constraint(m,  - x25 + x29 >= -2)

@constraint(m,  - x26 + x30 >= -3)

@constraint(m,  - x27 + x29 >= -1)

@constraint(m,  - x28 + x30 >= -2)
